src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -308,6 +292,9 @@ public void runDevAppServer1Command(List<String> jvmArgs, List<String> args,
     command.addAll(args);
 
     logCommand(command);
+
+    Map<String, String> environment = Maps.newHashMap();
+    environment.put("JAVA_HOME", javaBinaryPath.getParent().getParent().toString());
```

Comment:
It would attempt to invoke Java using its symlink.
I can't think of any issues that could arise from that.